examples/blogs-videos/mojo_dunder/DunderArray.mojo

```diff
@@ -0,0 +1,374 @@
+                    self._ptr.simd_load[simd_width](idx[0])|other._ptr.simd_load[simd_width](idx[0]))
+            elementwise[1, simdwidthof[dtype](), wrapper](self.numel)
+            return new_array
+        else:
+            print('Error: You can only AND int or bool arrays')
+            return self
+
+    fn __ior__(inout self, other: Self):
+        self = self.__or__(other)
+
+    fn __xor__(self, other: Self) -> Self:
+        if ~_mlirtype_is_eq[Scalar[dtype], Float64]():
+            let new_array = Self(self.numel)
+            @parameter
+            fn wrapper[simd_width:Int,rank:Int=1](idx: StaticIntTuple[rank]):
+                new_array._ptr.simd_store[simd_width](idx[0], 
+                    self._ptr.simd_load[simd_width](idx[0])^other._ptr.simd_load[simd_width](idx[0]))
+            elementwise[1, simdwidthof[dtype](), wrapper](self.numel)
+            return new_array
+        else:
+            print('Error: You can only AND int or bool arrays')
+            return self
+
+    fn __ixor__(inout self, other: Self):
+        self = self.__xor__(other)
+
+    fn __len__(self) -> Int:
+        return self.numel
+
+    fn __int__(self) -> Int:
+        return self.numel
+
+    fn __bool__(self) -> Bool:
+        return self.dtype == DType.bool
+
+    fn __str__(self) -> String:
+        var printStr:String = "["
+        var prec:Int=4
+        for i in range(self.numel):
+            var val = self[i]
+            @parameter 
+            if _mlirtype_is_eq[Scalar[dtype], Float64]():
+                var s: String = ""
+                let int_str: String
+                int_str = String(math.trunc(val).cast[DType.int32]())
+                if val < 0.0:
+                    val = -val
+                let float_str: String
+                if math.mod(val,1)==0:
+                    float_str = "0"
+                else:
+                    float_str = String(math.mod(val,1))[2:prec+2]
+                s = int_str+"."+float_str               
+                if i==0:
+                    printStr+=s
+                else:
+                    printStr+="  "+s
+            else:
+                if i==0:
+                    printStr+=str(val)
+                else:
+                    printStr+="  "+str(val)
+
+        printStr+="]\n"
+        printStr+="Length:"+str(self.numel)+","+" DType:"+str(dtype)
+        return printStr
+    
+    fn to_numpy(self) raises->PythonObject:
+        var np = Python.import_module("numpy")
+        var np_arr = np.zeros(self.numel)
+        var npArrayPtr = DTypePointer[dtype](
+        __mlir_op.`pop.index_to_pointer`[
+            _type = __mlir_type[`!kgen.pointer<scalar<`, dtype.value, `>>`]
+        ](
+            SIMD[DType.index,1](np_arr.__array_interface__['data'][0].__index__()).value
+        )
+    )
+        memcpy(npArrayPtr, self._ptr, self.numel)
+        return np_arr
+
+    fn _reduce_sum(self) -> Self:
```

Comment:
Since you use this function almost exclusively `xxx._reduce_sum()[0]`, wouldn't it be better if the return type is `Scalar[dtype]`?